accelerator/mps_accelerator.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import os
+import pkgutil
+import importlib
+
+from .abstract_accelerator import DeepSpeedAccelerator
+
+# During setup stage torch may not be installed, pass on no torch will
+# allow op builder related API to be executed.
+try:
+    import torch.mps
+except ImportError:
+    pass
+
+
+class MPS_Accelerator(DeepSpeedAccelerator):
+    def __init__(self):
+        self._name = "mps"
+        self._communication_backend_name = None
+
+    def is_synchronized_device(self):
+        return False
+
+    # Device APIs
+
+        def device_name(self, device_index=None):
+            if device_index == None:
+                return 'mps'
+            return 'mps:{}'.format(device_index)
```

Comment:
There seems to be an extra indent here, i think we can merge this after this fix though